Docs/src/ogre-shadows.md

```diff
@@ -397,15 +400,386 @@ Non-PCF block. Note that after doing this, the uSTexWidth and
+```
+
+Another example, using LiSPSM Camera Setup:
+```cpp
+Ogre::ShadowCameraSetupPtr shadowCameraSetup = Ogre::LiSPSMShadowCameraSetup::create();
+mSceneMgr->setShadowCameraSetup(shadowCameraSetup);
+```
+
+For big scenes with directional lights one of the better performing Shadow Camera Setups is PSSM.
+A PSSM shadow system uses multiple shadow maps per light and maps each texture into a region of space, 
+progressing away from the camera. As such it is most appropriate for directional light setups.
+A more in depth explanation can be found in the wiki: [Parallel Split Shadow Mapping](https://wiki.ogre3d.org/Parallel+Split+Shadow+Mapping)
+
+```cpp
+// General scene setup
+mSceneMgr->setShadowTechnique(Ogre::SHADOWTYPE_TEXTURE_ADDITIVE_INTEGRATED);
+mSceneMgr->setShadowTextureCasterMaterial(Ogre::MaterialManager::getSingletonPtr()->getByName("PSSMCaster"));
+mSceneMgr->setShadowTextureSelfShadow(true);
+mSceneMgr->setShadowFarDistance(3000);
+// 3 textures per directional light (PSSM)
+mSceneMgr->setShadowTextureCountPerLightType(Ogre::Light::LT_DIRECTIONAL, 3);
+mSceneMgr->setShadowTextureCount(3);
+mSceneMgr->setShadowTextureConfig(0, 2048, 2048, Ogre::PF_FLOAT16_RGB);
+mSceneMgr->setShadowTextureConfig(1, 1024, 1024, Ogre::PF_FLOAT16_RGB);
+mSceneMgr->setShadowTextureConfig(2, 512, 512, Ogre::PF_FLOAT16_RGB);
+
+Ogre::PSSMShadowCameraSetup* pssmSetup = new Ogre::PSSMShadowCameraSetup(); //static_cast<Ogre::PSSMShadowCameraSetup*>(shadowCameraSetup->getPointer());
+pssmSetup->setSplitPadding(1);
+pssmSetup->calculateSplitPoints(3, 1, mSceneMgr->getShadowFarDistance());
+pssmSetup->setOptimalAdjustFactor(0, 2);
+pssmSetup->setOptimalAdjustFactor(1, 1);
+pssmSetup->setOptimalAdjustFactor(2, 0.5);
+mSceneMgr->setShadowCameraSetup(Ogre::ShadowCameraSetupPtr(pssmSetup));
+```
+
+The Shadow Caster Vertex and Fragment programs are the same as the regular shadow mapping techniques.
+
+But some changes have to be made to the shaders of the Shadow Receiver as well as the program definition, because now we are sending three shadow map splits (in this example).
+
+Material definition:
+```
+material PSSMShadowReceiver
+{
+    technique default
+    {
+        pass
+        {
+            vertex_program_ref PSSMShadowReceiverVP {}
+            fragment_program_ref PSSMShadowReceiverFP {}
+
+            texture_unit ShadowMap0
+            {
+                tex_address_mode clamp
+				tex_border_colour 1.0 1.0 1.0 1.0
+				content_type shadow
+                filtering none
+            }
+
+            texture_unit ShadowMap1
+            {
+                tex_address_mode clamp
+				tex_border_colour 1.0 1.0 1.0 1.0
+				content_type shadow
+                filtering none
+            }
+
+            texture_unit ShadowMap2
+            {
+                tex_address_mode clamp
+				tex_border_colour 1.0 1.0 1.0 1.0
+				content_type shadow
+                filtering none
+            }
+        }
+    }
+}
+```
+
+Program definition:
```

Comment:
these shaders are a bad example... Ideally one would only show how to sample the shadow textures, however
- there is vertex lighting mixed in
- there is depth biasing with magic numbers (should not be done that way)
- there is PCF filtering which requires `inverse_texture_size`, but is #defined out

Reply:
But this shader is based on the one from the samples...

The only change is to make it work with PSSM.

Reply:
ah, I see.. the sample shader is no good either - I just did not get around to refactor the Shadows sample.

ok, then lets leave it like that and I will refactor all at once later